test/e2e/scenarios/ciliumeventobserver/validate-ciliumeventobserver-metric.go

```diff
@@ -0,0 +1,89 @@
+package ciliumeventobserver
+
+import (
+	"fmt"
+	"log"
+
+	prom "github.com/microsoft/retina/test/e2e/framework/prometheus"
+)
+
+var (
+	dropCountMetricName = "hubble_drop_total"
+	tcpFlagsMetricName  = "hubble_tcp_flags_total"
+	flowsMetricName     = "hubble_flows_processed_total"
+)
+
+const (
+	destinationKey = "destination"
+	sourceKey      = "source"
+	protcolKey     = "protocol"
+	reasonKey      = "reason"
+	directionKey   = "direction"
+)
+
+type CEODropMetric struct {
+	PortForwardedHubblePort string
+	Source                  string
+	Protocol                string
+	Reason                  string
+	Direction               string
+}
+
+type CEOFlowsAndTCPMetrics struct {
+	PortForwardedHubblePort string
+	Source                  string
+}
+
+func (v *CEODropMetric) Run() error {
+	promAddress := fmt.Sprintf("http://localhost:%s/metrics", v.PortForwardedHubblePort)
+
+	metric := map[string]string{
+		directionKey: v.Direction, reasonKey: PolicyDenied,
+	}
+
+	err := prom.CheckMetric(promAddress, dropCountMetricName, metric)
+	if err != nil {
+		return fmt.Errorf("failed to verify prometheus metrics %s: %w", dropCountMetricName, err)
+	}
+
+	log.Printf("found metrics matching %+v\n", metric)
+	return nil
+}
+
+func (v *CEODropMetric) Prevalidate() error {
+	return nil
+}
+
+func (v *CEODropMetric) Stop() error {
+	return nil
+}
+
+// Flows
+func (v *CEOFlowsAndTCPMetrics) Run() error {
+	promAddress := fmt.Sprintf("http://localhost:%s/metrics", v.PortForwardedHubblePort)
+
+	metric := map[string]string{}
```

Comment:
will only check the presence of PortForwardedHubblePort, we want to also check the attributes